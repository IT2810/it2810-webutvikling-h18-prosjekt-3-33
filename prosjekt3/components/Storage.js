import React from 'react';
import {
    AsyncStorage,
} from 'react-native';


/* Format of the storage:


{ CONTACTS:   [ { name: "Ulrik", age: 22, address: "Bakklandet"  },
                { name: "Sondre", age: 24, address: "Ila"        },
                { name: "Jørgen", age: 95, address: "Møhlenberg" }, 
                ....
              ],
  

  TASKS:      [ { title: "Meeting", date: "22-12-1995" },
                { title: "Dinner", date: "12-7-2995" },

                ...

              ]

  GOALS:      [ {title: "Stop smoking", duration: 5, start: "22-12-1995"}

              ] 
}





*/