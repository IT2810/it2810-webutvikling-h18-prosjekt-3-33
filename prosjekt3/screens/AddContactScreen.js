import React from 'react';
import { Alert, TextInput,StyleSheet, View} from 'react-native';
import { Constants } from 'expo';


export default class AddContactScreen extends React.Component {
    state = {
        name: '',
        email: '',
        phone: ''
      };

    render() {
  
      return (
          <View>
            <TextInput
                style={styles.input}
                value={this.state.name}
                onChangeText={name => this.setState({name})}
                ref={ref => {this._nameInput = ref}}
                placeholder="Full Name"
                autoFocus={true}
                autoCapitalize="words"
                autoCorrect={true}
                keyboardType="default"
                returnKeyType="next"
                onSubmitEditing={this._next}
                blurOnSubmit={false}
            />
            <TextInput
                style={styles.input}
                value={this.state.email}
                onChangeText={email => this.setState({email})}
                placeholder="email@example.com"
                />
            <TextInput
                style={styles.input}
                value={this.state.phone}
                onChangeText={phone => this.setState({phone})}
                keyboardType="default"
                placeholder="Phone number"
                onSubmitEditing={this._submit}
                
            />
          </View>
  
      );
    }
      
      _submit = () => {
        Alert.alert(`Welcome, ${this.state.name}! Confirmation email has been sent to ${this.state.email}
        and your phone number is ${this.state.phone}`);

      };
  }

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#ecf0f1',
    },
    header: {
      paddingTop: 20 + Constants.statusBarHeight,
      padding: 20,
      backgroundColor: '#336699',
    },
    description: {
      fontSize: 14,
      color: 'white',
    },
    input: {
      margin: 20,
      marginBottom: 0,
      height: 34,
      paddingHorizontal: 10,
      borderRadius: 4,
      borderColor: '#ccc',
      borderWidth: 1,
      fontSize: 16,
    },
  });