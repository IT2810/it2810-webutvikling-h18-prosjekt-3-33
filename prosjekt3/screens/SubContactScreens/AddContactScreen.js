import React from 'react';
import { Alert, TextInput,StyleSheet, View, Button} from 'react-native';
import { Constants } from 'expo';


export default class AddContactScreen extends React.Component {
    state = {
        name: '',
        email: '',
        phone: ''
      };

    render() {
  
      return (
          <View>
            <TextInput
                style={styles.input}
                value={this.state.name}
                onChangeText={name => this.handleNameChange(name)}
                ref={ref => {this._nameInput = ref}}
                placeholder="Full Name"
                autoCapitalize="words"
                keyboardType="default"
                returnKeyType="next"
                onSubmitEditing={this._next}
                blurOnSubmit={false}
            />
            <TextInput
                style={styles.input}
                value={this.state.email}
                onChangeText={email => this.handleEmailChange(email)}
                placeholder="email@example.com"
                />
            <TextInput
                style={styles.input}
                value={this.state.phone}
                onChangeText={phone => this.handleNumberChange(phone)}
                keyboardType="default"
                placeholder="Phone number"
            />

            <Button title="Add contact" onPress={this._submit}></Button>
          </View>

          
  
      );
    }
      
      _submit = () => {
        Alert.alert("Added contact: ", this.state.name);
        this.props.addContact(this.state.name, this.state.email, this.state.phone);

      };

      handleNameChange = (name) => {
        this.setState({name})
      }
      handleEmailChange = (email) => {
        this.setState({email})
      }
      handleNumberChange = (number) => {
        this.setState({number})
      }
  }

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#ecf0f1',
    },
    header: {
      paddingTop: 20 + Constants.statusBarHeight,
      padding: 20,
      backgroundColor: '#336699',
    },
    description: {
      fontSize: 14,
      color: 'white',
    },
    input: {
      margin: 20,
      marginBottom: 0,
      height: 34,
      paddingHorizontal: 10,
      borderRadius: 4,
      borderColor: '#ccc',
      borderWidth: 1,
      fontSize: 16,
    },
  });