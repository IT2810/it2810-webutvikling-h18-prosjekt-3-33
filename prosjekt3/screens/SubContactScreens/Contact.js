import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet, Button} from 'react-native';




const Contact = ({name}) => 
        <View>
        <TouchableOpacity onPress={() => alert(name)}>
                <Text style={styles.item}>
                {name}
                </Text>
        </TouchableOpacity>
        </View>;

export default Contact;

  const styles = StyleSheet.create({
    item: {
      padding: 10,
      fontSize: 18,
      height: 44,
    }
  }); 