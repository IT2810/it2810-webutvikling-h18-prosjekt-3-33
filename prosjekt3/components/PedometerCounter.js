import React from 'react';
import Expo from 'expo';
import { Pedometer } from 'expo';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import * as Storage from './Storage';
import Colors from '../constants/Colors';

//Pedometer component used by StepGoal to get step data from core motion (iOS) or
//Google Fit (Android)
export default class PedometerCounter extends React.Component {
  state = {
    isPedometerActive: "checking",
    pastStepCount: 0,
    currentStepCount: 0,
    stepGoal: 0,
  };
  //Subscribe to pedometer count and setting state from props (from StepGoal)
  componentDidMount() {
    this._subscribe();
    this.setState({
      stepGoal: this.props.stepGoal
    })
  }
  //Unsubscribe from pedometer count on unmount
  componentWillUnmount() {
    this._unsubscribe();
  }
  //State changes on new props from StepGoal
  componentDidUpdate(prevProps, prevState){
    if(this.props.stepGoal != prevProps.stepGoal){
      this.setState({stepGoal: this.props.stepGoal})
    }
  }
//Subscribing to Core Motion and Google FIt
  _subscribe = () => {
    this._subscription = Pedometer.watchStepCount(result => {
      this.setState({
        currentStepCount: result.steps
      });
    });

    Pedometer.isAvailableAsync().then(
      result => {
        this.setState({
          isPedometerActive: String(result)
        });
      },
      error => {
        this.setState({
          isPedometerActive: "Could not get isPedometerActive: " + error
        });
      }
    );

    const end = new Date();
    const start = new Date();
    start.setDate(end.getDate() - 1);
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ pastStepCount: result.steps });
      },
      error => {
        this.setState({
          pastStepCount: "Could not get stepCount: " + error
        });
      }
    );
  };

  _unsubscribe = () => {
    this._subscription && this._subscription.remove();
    this._subscription = null;
  };

  render() {
    //Renders a progress bar based on progress / goal relationship
    var totalProgress = this.state.pastStepCount + this.state.currentStepCount;
    var goal = this.state.stepGoal;
    return (
      <View style={styles.container}>
        <View style={styles.textContainer}>
          <Text>
            {totalProgress} out of {goal}
          </Text>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyle(totalProgress, goal)}>
          </View>
        </View>

      </View>
    );
  }
}
//Function for calculating the progressbar colors and width
progressBarStyle = function(progress, goal){
  let progressWidth = 0;
  let backgroundColor = '#000';
  let progressGoalRelation = progress/goal;
  if(progressGoalRelation>=1){
    progressWidth = 1;
    backgroundColor = '#75CDA8';
  }
  if (progressGoalRelation < 1 && progressGoalRelation > 0.4){
    progressWidth = progressGoalRelation;
    backgroundColor = '#EDBF2C';
  }
  if(progressGoalRelation<=0.4){
    progressWidth = progressGoalRelation;
    backgroundColor = '#DB0200';
  }
  return{
    flex: progressWidth,
    backgroundColor: backgroundColor,
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    marginTop: 15,
    alignItems: "center",
    justifyContent: "center"
  },
  progressBar: {
    flex: 1,
    flexDirection: 'row',
    width: 300,
    height: 20,
    backgroundColor: '#eee',
  },
  textContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    width: 300,
  }
});

Expo.registerRootComponent(PedometerCounter);
