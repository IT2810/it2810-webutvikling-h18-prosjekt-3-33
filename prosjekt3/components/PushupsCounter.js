import React from 'react';
import Expo from 'expo';
import { StyleSheet, Text, View, TouchableOpacity, Button } from 'react-native';
import * as Storage from './Storage';
import Colors from '../constants/Colors';

export default class PushupsCounter extends React.Component {
  state = {
    pastPushupsCount: 0,
    currentPushupsCount: 0,
    pushupsGoal: 5
  };
  componentDidMount() {
    this.setState({
      pushupsGoal: this.props.pushupsGoal,
    })
    Storage.getGoals().then(goals =>{
      if(goals.hasOwnProperty('pushupsCount')){
        this.setState({pastPushupsCount: goals.pushupsCount})
      }else{
        goals.pushupsCount = 0;
        Storage.storeGoals(goals);
      }
    });
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.pushupsGoal != prevProps.pushupsGoal){
      this.setState({pushupsGoal: this.props.pushupsGoal})
    }
    if(this.state != prevState){
      Storage.getGoals().then(goals =>{
        goals.pushupsCount = this.state.pastPushupsCount + this.state.currentPushupsCount;
        Storage.storeGoals(goals);
      });
    }
  }

  decrementCounter(totalProgress){
    this.setState({
      currentPushupsCount: (totalProgress-1 < 0)
        ? this.state.currentPushupsCount
        : this.state.currentPushupsCount-1
    })
  }

  incrementCounter(){
    this.setState({currentPushupsCount: this.state.currentPushupsCount+1})
  }

  progressBarStyle(progress, goal){
    let progressWidth = 0;
    let backgroundColor = '#000';
    let progressGoalRelation = progress/goal;
    if(progressGoalRelation>=1){
      progressWidth = 1;
      backgroundColor = '#75CDA8';
    }
    if (progressGoalRelation < 1 && progressGoalRelation > 0.4){
      progressWidth = progressGoalRelation;
      backgroundColor = '#EDBF2C';
    }
    if(progressGoalRelation<=0.4){
      progressWidth = progressGoalRelation;
      backgroundColor = '#DB0200';
    }
    return{
      flex: progressWidth,
      backgroundColor: backgroundColor,
    }
  }

  render() {
    var totalProgress = this.state.pastPushupsCount + this.state.currentPushupsCount;
    var goal = this.state.pushupsGoal;
    return (
      <View style={styles.container}>
        <View style={styles.textContainer}>
          <View style={styles.minusButtonContainer}>
            <Button
              title="-"
              onPress={() => this.decrementCounter(totalProgress)} />
          </View>
          <Text>
            {totalProgress} out of {goal}
          </Text>
          <View style={styles.buttonContainer}>
            <Button
              title="+"
              onPress={() => this.incrementCounter()}
              style={styles.plusButton}/>
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={this.progressBarStyle(totalProgress, goal)}>
          </View>
        </View>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    marginTop: 15,
    alignItems: "center",
    justifyContent: "center"
  },
  progressBar: {
    flex: 1,
    flexDirection: 'row',
    width: 300,
    height: 20,
    backgroundColor: '#eee',
    marginTop: 20,
  },
  textContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: 300,
  },
  plussButton: {
    backgroundColor: '#2db747',
    fontSize: 25,
  },
  buttonContainer: {
    backgroundColor: '#15c129',
    height: 40,
    width: 40,
  },
  minusButtonContainer: {
    backgroundColor: '#c11537',
    height: 40,
    width: 40,
  }
});
