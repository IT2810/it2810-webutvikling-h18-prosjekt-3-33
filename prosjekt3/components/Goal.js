import React from 'react';
import {
  View, Text, StyleSheet, ScrollView

} from 'react-native';
import Colors from '../constants/Colors.js';
import PedometerCounter from './PedometerCounter';

export default class Goal extends React.Component {
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of steps</Text>
          <PedometerCounter />
        </View>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Hours of studing</Text>
        </View>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of movies watched</Text>
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.unfinishedGoalColor,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
