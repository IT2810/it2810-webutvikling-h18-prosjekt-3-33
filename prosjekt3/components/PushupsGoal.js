import React from 'react';
import {
  View, Text, StyleSheet, ScrollView, Button, TouchableOpacity

} from 'react-native';
import Colors from '../constants/Colors.js';
import PushupsCounter from './PushupsCounter';

export default class PushupsGoal extends React.Component {
  state = {
    pushupsGoal: 0,
  }
  componentDidMount(){
    this.setState({pushupsGoal: this.props.pushupsGoal})
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.studyGoal != prevProps.studyGoal){
      this.setState({pushupsGoal: this.props.pushupsGoal})
    }
  }
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of pushups today </Text>
          <PushupsCounter pushupsGoal={this.props.pushupsGoal} />
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.goalComponentOne,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
