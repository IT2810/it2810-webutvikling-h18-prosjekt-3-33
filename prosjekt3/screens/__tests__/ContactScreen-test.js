import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ContactScreen from '../ContactsScreen';
import Contact from '../SubContactScreens/Contact';
import AddContactScreen from '../SubContactScreens/AddContactScreen';

import * as Storage from '../../components/Storage';


describe("ContactScreen Testing", () => {
    
    it('renders the CalendarScreen', async () => {
        const tree = renderer.create(<ContactScreen />).toJSON();
        expect(tree).toMatchSnapshot();
      });

    it('render the Contact Components', async () => {
        let item = {
            name: 'Andreas',
            email: 'andi@andi.com',
            number: 41403912
        }
        const tree = renderer.create(<Contact item={item} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should change state in sections, and be equal to', () => {
        let contactComponent = renderer.create(<ContactScreen />).getInstance()
        contactComponent.addContact('Jørgen','Jorgensta@gmail.com',41403910);

        let section = {
            title: 'J',
            data: [{
              name: 'Jørgen',
              email: 'Jorgensta@gmail.com',
              number: 41403910
            }],
          }
        expect(contactComponent.state.sections).toEqual([section])

    })

    it('should show set visible Modal, and match snapshot after invoke', () => {
        let contactComponent = renderer.create(<ContactScreen />)
        contactComponent.getInstance().setModalVisible(true)
        expect(contactComponent.getInstance().state.modalVisible).toEqual(true)
        const afterTree = contactComponent.toJSON();
        expect(afterTree).toMatchSnapshot();

    })
})

describe("AddContact screen testing", () => {
    it('renders the addContactScreen correctly', async () => {
        const tree = renderer.create(<AddContactScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("should update addContact state name correctly", async () => {
        const addContactComponent = renderer.create(<AddContactScreen />).getInstance();
        addContactComponent.handleNameChange('jørgen');
        expect(addContactComponent.state.name).toEqual('jørgen')
    })

    it("should update addContact state Email correctly", async () => {
        const addContactComponent = renderer.create(<AddContactScreen />).getInstance();
        addContactComponent.handleEmailChange('jorg@jorg.com');
        expect(addContactComponent.state.email).toEqual('jorg@jorg.com')
    })

    it("should update addContact state Phonecorrectly", async () => {
        const addContactComponent = renderer.create(<AddContactScreen />).getInstance();
        addContactComponent.handleNumberChange('41403910');
        expect(addContactComponent.state.number).toEqual('41403910')
    })


    it('AddContact method should be called' , async () => {
        const addContact = jest.fn();

        let testState = {
            name: 'Jørgen',
            email: 'Jorgensta@gmail.com',
            number: 41403910
        }
        
        const addContactScreen = renderer.create(<AddContactScreen state={testState} addContact={addContact} />).getInstance();

        addContactScreen._submit()

        expect(addContact).toHaveBeenCalled();
    })

})