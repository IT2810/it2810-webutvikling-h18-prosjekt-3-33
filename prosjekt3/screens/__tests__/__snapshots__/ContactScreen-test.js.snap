// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContact screen testing renders the addContactScreen correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    blurOnSubmit={false}
    keyboardType="default"
    onChangeText={[Function]}
    onSubmitEditing={undefined}
    placeholder="Full Name"
    returnKeyType="next"
    style={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 34,
        "margin": 20,
        "marginBottom": 0,
        "paddingHorizontal": 10,
      }
    }
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="email@example.com"
    style={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 34,
        "margin": 20,
        "marginBottom": 0,
        "paddingHorizontal": 10,
      }
    }
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Phone number"
    style={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 34,
        "margin": 20,
        "marginBottom": 0,
        "paddingHorizontal": 10,
      }
    }
    value=""
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add contact
      </Text>
    </View>
  </View>
</View>
`;

exports[`ContactScreen Testing render the Contact Components 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "height": 44,
          "padding": 10,
        }
      }
    >
      Andreas
    </Text>
  </View>
</View>
`;

exports[`ContactScreen Testing renders the CalendarScreen 1`] = `null`;

exports[`ContactScreen Testing should show set visible Modal, and match snapshot after invoke 1`] = `null`;
